### DISCUSSION TITLE
Testing form actions

### DISCUSSION THREAD

**Discussion by nasegplus** [filamentphp/filament#14494] [Ideas]
Couldn't see this in the docs. Given that a livewire component is set up with a custom form and action as follows:

```php
public function form(Form $form): Form
{
    return $form->schema([
        TextInput::make('name')
            ->required(),
        Actions::make([
            Action::make('action')
                ->action(/** doing something **/)
    ]);
}
```

With the following test (which fails) I can't find a way to pass the formName to callAction order to test the action:

```php
livewire(MyPage::class)
    ->fillForm([
        'name' => 'something'
    ])
    ->callAction('action');
```
This results in Failed asserting that an action with name [action] exists on the [App\Filament\Pages\MyPage] page. Does such functionality exist?

Here a component name should be provided but we don't have one.

```php
livewire(MyPage::class)
    ->fillForm([
        'name' => 'something'
    ])
    ->assertFormComponentActionExists('component','action')
}
```
We need to solve this.

**Comment by leandrocfe** [comment]
https://github.com/filamentphp/filament/discussions/14493

ref: https://filamentphp.com/docs/3.x/forms/testing#actions
